src/App.tsx

```diff
@@ -1,8 +1,104 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import TodoList from './components/TodoList';
+import { Todo, User } from './interfaces';
+import './App.scss';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [show, setShow] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  async function start() {
+    setLoading(true);
+    const load = await Promise.all([
+      fetch('https://mate.academy/students-api/todos')
+        .then(response => response.json())
+        .then(response => {
+          setTodos(response.data);
+        }),
+      fetch('https://mate.academy/students-api/users')
+        .then(response => response.json())
+        .then(response => {
+          setUsers(response.data);
+        }),
+    ]);
+
+    if (load) {
+      setShow(true);
+    }
+  }
+
+  const onSortbyTitle = () => {
+    const sorted = [...todos].sort((a: Todo, b: Todo) => (a.title).localeCompare(b.title));
+
+    setTodos(sorted);
+  };
+
+  const onSortbyCompleted = () => {
+    const sorted = [...todos].sort((a: Todo, b: Todo) => +a.completed - +b.completed);
+
+    setTodos(sorted);
+  };
+
+  const onSortbyName = () => {
+    const sorted = [...todos].sort((a: Todo, b: Todo) => {
+      const name1: User | undefined = users.find((user: User) => user.id === a.userId);
+      const name2: User | undefined = users.find((user: User) => user.id === b.userId);
```

Comment:
```suggestion
      const userA = users.find((user) => user.id === a.userId) as User;
      const userB = users.find((user) => user.id === b.userId) as User;
```
это юзеры а не имена, тут не нужно описывать типы потому что юзеров ты описал еще в стейте